frontend/public/components/chargeback.tsx

```diff
@@ -317,25 +243,21 @@ class ReportData extends React.Component<ReportDataProps, ReportDataState> {
 
   render() {
     const {obj} = this.props;
-    const phase = _.get(obj, ['status', 'phase']);
-
-    const {data, reduceBy, sortBy, orderBy, keys, rows, maxValues, totals, inFlight, error} = this.state;
+    const {data, sortBy, orderBy, keys, rows, inFlight, error} = this.state;
 
     let dataElem = <MsgBox title="No Data" detail="Report not finished running." />;
     if (inFlight) {
       dataElem = <div className="row"><div className="col-xs-12 text-center"><LoadingInline /></div></div>;
     } else if (error) {
-      dataElem = <LoadError label="Report" message={error} />;
-    } else if (phase === 'Finished') {
-      if (data) {
-        dataElem = <DataTable sortBy={sortBy} orderBy={orderBy} keys={keys} rows={rows} maxValues={maxValues} totals={totals} applySort={(sb, func, ob) => this.applySort(sb, func, ob)} />;
+      dataElem = <LoadError label="Report" message={_.get(error, 'json.error') || error.message} />;
+    } else if (data) {
+      if (data.error) {
+        dataElem = <LoadError label="Report" message={data.error} />;
       } else {
-        dataElem = <MsgBox title="No Data" detail="" />;
+        dataElem = <DataTable sortBy={sortBy} orderBy={orderBy} keys={keys} rows={rows}
+          applySort={(sb, func, ob) => this.applySort(sb, func, ob)} />;
       }
```

Comment:
Don't we need a final `else` to show the empty state message if `data` is not defined?

Reply:
No, `dataElem` defined initially, before IF's, as a 'No Data' MsgBox